packages/engine-formula/src/engine/dependency/dependency-tree.ts

```diff
@@ -234,19 +209,32 @@ export class FormulaDependencyTree extends Disposable {
     pushChildren(tree: FormulaDependencyTree) {
         this.children.push(tree);
         tree._pushParent(this);
-        this._childIds.add(tree.id);
-    }
-
-    hasChild(tree: FormulaDependencyTree) {
-        return this._childIds.has(tree.id);
     }
 
     /**
      * Add the range corresponding to the current ast node.
      * @param range
      */
-    pushRangeList(range: IUnitRangeWithToken) {
-        this.rangeList.push(range);
+    pushRangeList(ranges: IUnitRange[]) {
+        this.rangeList.push(...ranges);
+    }
+
+    hasChildren(tree: FormulaDependencyTree) {
+        return this.children.includes(tree);
```

Comment:
please use map or set instead of loop all child

Reply:
Done